magic/malloc_allocator.hpp

```diff
@@ -0,0 +1,110 @@
+// This file contains two things:
```

Comment:
Among other things, the `magic` directly is going to interact poorly with VS.  I would organize things differently here:

1. A file `base/malloc_allocator.hpp` that declares and defines the class `MallocAllocator` in namespace `principia::base`.  This one is not magic in any way, and should be included in the usual manner.
2. A file `base/macos_allocator_replacement.hpp` that has all the stuff declared in `principia::std`.  This is the one that is magically included.  For safety, it should have preprocessor directives that ensure that it's never pulled for other platforms:
```c++
#if !OS_MACOSX
#error Only include for macOS
#endif 
... all the rest ...
```
You'll need to include `base/macros.hpp` in that second file.

Reply:
Done.